crates/aptos/src/account/key_rotation.rs

```diff
@@ -286,41 +283,66 @@ impl CliCommand<AccountAddress> for LookupAddress {
-            .await?
-            .into_inner();
-
-        let table_handle = originating_resource.address_map.handle;
-
-        // The derived address that can be used to look up the original address
         // TODO: This command needs to support multi-ed25519
-        let address_key = AuthenticationKey::ed25519(&self.public_key()?).derived_address();
-        match rest_client
-            .get_table_item_bcs(
-                table_handle,
-                "address",
-                "address",
-                address_key.to_hex_literal(),
-            )
-            .await
-        {
-            Ok(inner) => Ok(inner.into_inner()),
-            Err(RestError::Api(AptosErrorResponse {
+        let address_key = account_address_from_public_key(&self.public_key()?);
+        Ok(lookup_address(&rest_client, address_key, true).await?)
+    }
+}
+
+pub async fn lookup_address(
+    rest_client: &Client,
+    address_key: AccountAddress,
+    must_exist: bool,
+) -> Result<AccountAddress, RestError> {
+    let originating_resource: OriginatingResource = rest_client
+        .get_account_resource_bcs(CORE_CODE_ADDRESS, "0x1::account::OriginatingAddress")
+        .await?
+        .into_inner();
+
+    let table_handle = originating_resource.address_map.handle;
+
+    // The derived address that can be used to look up the original address
+    match rest_client
+        .get_table_item_bcs(
+            table_handle,
+            "address",
+            "address",
+            address_key.to_hex_literal(),
+        )
+        .await
+    {
+        Ok(inner) => Ok(inner.into_inner()),
+        Err(RestError::Api(AptosErrorResponse {
+            error:
+                AptosError {
+                    error_code: AptosErrorCode::TableItemNotFound,
+                    ..
+                },
+            ..
+        })) => {
+            // If the table item wasn't found, let's at least check if the account exists
+            // It won't be in the table if it wasn't rotated, then return the derived account address
+            let result = rest_client.get_account_bcs(address_key).await;
+
+            if let Err(RestError::Api(AptosErrorResponse {
                 error:
                     AptosError {
-                        error_code: AptosErrorCode::TableItemNotFound,
+                        error_code: AptosErrorCode::AccountNotFound { .. },
                         ..
                     },
                 ..
-            })) => {
-                // If the table item wasn't found, let's at least check if the account exists
-                // It won't be in the table if it wasn't rotated, then return the derived account address
-                rest_client.get_account_bcs(address_key).await?;
-                Ok(address_key)
-            },
-            Err(err) => Err(err)?,
-        }
+            })) = result
+            {
+                if must_exist {
+                    return Err(result.unwrap_err());
```

Comment:
Could you just do `return result`?

Reply:
It's a different result of a response, and not an account address, so it can't

Reply:
Perhaps it'd be better to transform it into an anyhow result using `.context()?` instead.